connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/MySQLIntegrationSuite.scala

```diff
@@ -154,30 +154,43 @@ class MySQLIntegrationSuite extends DockerJDBCIntegrationSuite {
   test("SPARK-47462: Unsigned numeric types") {
     val df = sqlContext.read.jdbc(jdbcUrl, "unsigned_numbers", new Properties)
     val rows = df.head()
-    assert(rows.get(0).isInstanceOf[Short])
-    assert(rows.get(1).isInstanceOf[Integer])
-    assert(rows.get(2).isInstanceOf[Integer])
-    assert(rows.get(3).isInstanceOf[Long])
-    assert(rows.get(4).isInstanceOf[BigDecimal])
-    assert(rows.get(5).isInstanceOf[BigDecimal])
-    assert(rows.get(6).isInstanceOf[Double])
-    // Unlike MySQL, MariaDB seems not to distinguish signed and unsigned tinyint(1).
     val isMaria = jdbcUrl.indexOf("disableMariaDbDriver") == -1
     if (isMaria) {
+      assert(rows.get(0).isInstanceOf[Integer])
+      assert(rows.get(1).isInstanceOf[Long])
+      assert(rows.get(2).isInstanceOf[Integer])
+      assert(rows.get(3).isInstanceOf[BigDecimal])
```

Comment:
Could you describe the root cause of this change, @panbingkun ? For me, this looks suspicious like a breaking change.
```scala
- assert(rows.get(3).isInstanceOf[Long])
+ assert(rows.get(3).isInstanceOf[BigDecimal])
```

Reply:
the root cause of this change
- change:
  index[0], Short => Integer
  index[1, Integer => Long
  index[3], Long => Decimal

- Before - (mariadb jdbc driver: 2.7.12):
  <img width="1037" alt="image" src="https://github.com/apache/spark/assets/15246973/58a86548-718b-4115-8b4b-7a6dce2bbfcc">

- After - (mariadb jdbc driver: 3.4.0):
  <img width="1240" alt="image" src="https://github.com/apache/spark/assets/15246973/6d14d445-8985-48cf-96a9-dcd13b9a945b">

Reply:
We can also add compatible logic in `MySQLDialect##getCatalystType`. Do we need to do this in this PR?

Reply:
We can make changes as follows to keep this compatible:
<img width="899" alt="image" src="https://github.com/apache/spark/assets/15246973/6c3b04f3-b846-4b2b-84df-3134540cbd89">

If we need to do this, how about making related changes directly in this PR? Or a new seperated PR?

Reply:
> After - (mariadb jdbc driver: 3.4.0):

These mariadb SQL Data Type <-> JDBC Data Type mapping rules seem incorrect, do they have a statement why they made such change

Reply:
Let me try to find it.